Automattic/jetpack
# Update the post-list package post type availability

This PR ensures Thumbnails are only added to "Posts" and "Pages" because those areas are predictable and we don't want to mess up custom post type formatting by adding thumbnails.

This PR also limits the Share action to post types that support 'publicize', use the 'block editor,' and have the Share feature flag enabled. This prevents us from adding Share to post types when it doesn't make sense.

Unit tests have been added to test for these scenarios.

Fixes #21526

#### Changes proposed in this Pull Request:

* Thumbnails only available on Pages and Posts specifically.
* Share action only available on post types that support 'publicize', the 'block-editor', and have the Share feature flag enabled.

#### Jetpack product discussion
See issue.

#### Does this pull request change what data or activity we track or use?
No.

#### To run the unit tests
 - go the post-list folder "cd projects/packages/post-list"
 - run the command "composer test-php"

#### Testing instructions:
 - Apply this PR.
 - Enable `site_sticker_wpcom-post-list-enhancements` and `site_sticker_wpcom-gutenberg-republicize` stickers.
 - Thumbnails should only be visible on `wp-admin/edit.php?post_type=post` and `wp-admin/edit.php?post_type=page`
 - The Share action should only be visible on `wp-admin/edit.php?post_type=post`

This can be tested on a Jetpack site using Paul's method in the [comments here](https://github.com/Automattic/jetpack/pull/21577#pullrequestreview-793254297).

Or, you can test this on mock Atomic site by installing the WPCOMSH plugin on a local Jetpack install. Enable WPCOMSH by adding `define( 'IS_ATOMIC', true );` in `wpcomsh-loader.php`. Then mock the stickers in `tools/docker/mu-plugins/0-functions.php` or similar with the following code snippet:

```
final class Atomic_Persistent_Data {
	public function __get( $name ) {
		if ( 'site_sticker_wpcom-post-list-enhancements' === $name ) {
			return true;
		}

		if ( 'site_sticker_wpcom-gutenberg-republicize' === $name ) {
			return true;
		}
		return null;
	}
}
```
